src/plots/cartesian/axes.js

```diff
@@ -3801,19 +3827,7 @@ axes.drawLabels = function(gd, ax, opts) {
         ax._selections[cls] = tickLabels;
     }
 
-    var seq = [allLabelsReady];
-
-    // N.B. during auto-margin redraws, if the axis fixed its label overlaps
-    // by rotating 90 degrees, do not attempt to re-fix its label overlaps
-    // as this can lead to infinite redraw loops!
-    if(ax.automargin && fullLayout._redrawFromAutoMarginCount && prevAngle === 90) {
```

Comment:
I'm not sure if it is safe to remove this block?

Reply:
I removed this because of the first comment at the top of this PR. Have you checked that?

Reply:
TBH, this isn't even related to the feature I made. The discussion only started because I changed the check `prevAngle === 90` to also consider negative 90.
If you are unsure about it, I can also just revert those changes and leave it as it was before, so that you can properly figure out how the prevention of infinite redraws should really work.

Reply:
I went back to look at where this block was originally introduced, it was in https://github.com/plotly/plotly.js/pull/4216 back in v1.49.5 (2019). Adapting the [first codepen shown there](https://codepen.io/alanjmac/pen/oraKpR) to the latest release and to this PR, neither shows an infinite loop AFAICT, so that's great. But the current PR has some history-dependent behavior that the latest release does not.

Here's on the latest release: https://codepen.io/alexcjohnson/pen/YzBovrN - the angle starts 30 degrees, then changes to 90 degrees when we shrink the width, then goes back to 30 degrees when we go back to the original width (this is font-dependent, so it's possible on other machines the specific widths will need to be adjusted)

And here's on this branch: https://codepen.io/alexcjohnson/pen/abXgKad - note the angle stays at 90 degrees at the end. It doesn't go back down until you get wide enough to go back to 0 degrees.

Reply:
I have removed my changes for now, so the old behavior is restored.